sample_code/ep359/rev01/t.py

```diff
@@ -0,0 +1,20 @@
+
+
+try:
+    # code which may raise an exception
+    ...
+except (TypeError, AssertionError):
+    # code that runs when an exception occurs
+    ...
+except Exception as e:
+    # code that runs when an exception occurs
+    ...
+except BaseException:
+    # code that runs when an exception occurs
+    ...
+else:
+    # code that only runs if successful
+    ...
+finally:
+    # always runs independent of success
+    ...
```

Comment:
Reordered the `else` and `finally` clauses, because else can't be at the end. Python throws `SyntaxError`. :D

Reply:
yeah I made a mistake heh